### DISCUSSION TITLE
SMTP-only: Client host rejected

### DISCUSSION THREAD

**Discussion by dandln** [docker-mailserver/docker-mailserver#2842] [Q&A] [answered]
Hi,

I'm trying to set docker-mailserver up as an SMTP-only mail server to send emails from my NextCloud AIO instance. It's a fairly basic setup, I made the mailserver container join the NextCloud docker network to make sure that it can accept send requests from there and only from there (ports are not exposed to the internet).

My configuration for sending emails in NextCloud is to connect via SSL/TLS to 172.21.0.2 (the mailserver's container IP) on port 465, and the connection itself seems to be successful, however I can't get it to actually send any email. Mail logs show the following:
> Oct 19 23:53:30 mail postfix/smtps/smtpd[4994]: connect from nextcloud-aio-nextcloud.nextcloud-aio[172.21.0.9]
Oct 19 23:53:30 mail postfix/smtps/smtpd[4994]: Anonymous TLS connection established from nextcloud-aio-nextcloud.nextcloud-aio[172.21.0.9]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256
Oct 19 23:53:30 mail opendmarc[354]: ignoring connection from nextcloud-aio-nextcloud.nextcloud-aio
Oct 19 23:53:30 mail postfix/smtps/smtpd[4994]: NOQUEUE: reject: RCPT from nextcloud-aio-nextcloud.nextcloud-aio[172.21.0.9]: 554 5.7.1 <nextcloud-aio-nextcloud.nextcloud-aio[172.21.0.9]>: Client host rejected: Access denied; from=<_[configured_header_from_in_nextcloud]_> to=<_[test_recipient]_> proto=ESMTP helo=<_[email_domain]_>
Oct 19 23:53:30 mail postfix/smtps/smtpd[4994]: disconnect from nextcloud-aio-nextcloud.nextcloud-aio[172.21.0.9] ehlo=1 mail=1 rcpt=0/1 data=0/1 quit=1 commands=3/5

In this setup, NextCloud is not authenticating with a user, but I have also tried that and it pretty much yields the same result. Now the error message is pretty clear in what goes wrong, as a complete newbie to postfix I just don't really know how to fix this. Anyone got any ideas? :)

**Comment by casperklein** [comment, marked as answer]
Try port 25 instead, that should work.

If you haven't done so, you need to configure the PERMIT_DOCKER variable:

https://github.com/docker-mailserver/docker-mailserver/blob/b60b4015ab74d9a1b2cba47213919bfa8c2413fb/mailserver.env#L57-L69

This allows sending mails without authentication.

**Reply by dandln** [reply]
Using port 25 and no encryption works, though this obviously drops encryption, which... I guess is okay for just container to container communication. Will leave it at that for now. Thank you!

**Reply by casperklein** [reply]
You can use STARTTLS on port 25.

**Reply by dandln** [reply]
Yeah good point, not sure how I missed that haha. Thanks!
